\documentclass{article}


% if you need to pass options to natbib, use, e.g.:
%     \PassOptionsToPackage{numbers, compress}{natbib}
% before loading neurips_2022

% References (added by Sophia)
% https://www.overleaf.com/learn/latex/Bibliography_management_in_LaTeX
% https://en.wikibooks.org/wiki/LaTeX/Bibliography_Management
\usepackage[authordate,bibencoding=auto,strict,backend=biber,natbib]{biblatex-chicago}
\usepackage[english]{babel}
\addbibresource{References.bib}

\DefineBibliographyStrings{english}{%
  andothers = {et al.},
}

\usepackage[pdftex]{graphicx}

% ready for submission
\usepackage[final,nonatbib]{neurips_2022}


% to compile a preprint version, e.g., for submission to arXiv, add add the
% [preprint] option:
%     \usepackage[preprint]{neurips_2022}


% to compile a camera-ready version, add the [final] option, e.g.:
%     \usepackage[final]{neurips_2022}


% to avoid loading the natbib package, add option nonatbib:
%    \usepackage[nonatbib]{neurips_2022}


\usepackage[utf8]{inputenc} % allow utf-8 input
\usepackage[T1]{fontenc}    % use 8-bit T1 fonts
\usepackage{hyperref}       % hyperlinks
\usepackage{url}            % simple URL typesetting
\usepackage{booktabs}       % professional-quality tables
\usepackage{amsfonts}       % blackboard math symbols
\usepackage{nicefrac}       % compact symbols for 1/2, etc.
\usepackage{microtype}      % microtypography
\usepackage{xcolor}         % colors



\title{scVI enables effective scRNA-seq data preprocessing and highlights differences in HPSC transcriptional profiles of Aplastic Anemia patients and healthy donors}


% The \author macro works with any number of authors. There are two commands
% used to separate the names and addresses of multiple authors: \And and \AND.
%
% Using \And between authors leaves it to LaTeX to determine where to break the
% lines. Using \AND forces a line break at that point. So, if LaTeX puts 3 of 4
% authors names on the first line, and the last on the second line, try using
% \AND instead of \And before the third author name.


\author{%
Yegor Kuznetsov$^{1}$ \quad Sophia Jannetty$^{2}$\\
$^1$University of Washington Department of Computer Science\\
$^2$University of Washington Department of Biology\\
\texttt{yegor@uw.edu}\\
\texttt{jannetty@uw.edu}\\
}


\begin{document}


\maketitle


\begin{abstract}
  Aplastic anemia (AA) is a devastating illness that is challenging to study.
  Single-cell RNA-sequencing (scRNA-seq) datasets characterizing the transcriptional profiles of relevant cell types in AA patients have been published, but existing analyses of these datasets have been limited to clustering on dimensionalty-reduced data and performing differential expression analysis on seperate clusters.
  Here, we perform our own analysis on scRNA-seq data published by \citet{tonglin_single-cell_2022}.
  We begin by assessing the performance of a simple linear classifier model trained on the dimensionality-reduced data.
  We then use scVI to do something.
  We conclude that something.
\end{abstract}


\section{Introduction}

Aplastic anemia (AA) is a condition in which a patient\textquotesingle s bone marrow fails to form enough red blood cells, white blood cells, and platelets, resulting in pancytopenia \citep{young_aplastic_2018}.
Pathophysiological mechanisms of this disease include direct damage to bone marrow (commonly from chemotherapy), germ line loss-of-function mutations that interfere with blood-cell precursor DNA repair pathways, and autoimmune attack \citep{young_aplastic_2018}. 

Current options for treating AA fall within three categories, each of which attempts to address pancytopenia in a different way.
The first, bone marrow transplantation, aims to replace failing bone marrow with healthy donor marrow. 
This is a curative treatment, but is limited by the prevalence of graft-versus-host disease and by its reliance on tissue donors \citep{young_aplastic_2018}.
% Furthermore, the survival rate for adults over the age of 40 following bone marrow transplantation is only around 50\% \citep{giammarco_transplant_2018}.
The second, immunosuppression, aims to eliminate immune-mediated AA.
This category includes treatment with antilymphocyte globulin (combined with cyclosporine), which has a mild lymphocyte-depleting effect.
This treatment leads to improved blood production in about 66\% of patients \citep{bacigalupo_how_2017}, however 30\% to 60\% of these patients experience relapse that requires years of continued cyclosporine therapy \citep{scheinberg_activity_2012}.
Despite the success of immunosuppressant treatments, the mechanisms by which the immune system damages bone marrow remain unknown; the strongest evidence for immune-mediation of aplastic anemia is the effectiveness of immunosuppressant treatments \citep{young_aplastic_2018}.
The third treatment category, stem-cell stimulation, aims to promote stem-cell regeneration within the patient directly.
One such treatment involves administrating eltrombopag, a synthetic mimic of thrombopoietin (a hormone that stimulates the production of platelets).
This treatment has been shown in limited clinical trials to improve blood production in 80\% of treated patients, though the mechanism of this treatment remains unknown \citep{young_aplastic_2018}.

Despite the severity of this disease, the mechanisms underlying its onset and the treatment options remain obscure due to limitations in experimental techniques.
It is thought that T cells may target hematopoietic stem and progenitor cells (HSPCs), which are blood cell precursor cells, in patients with immune-mediated AA \citep{tonglin_single-cell_2022}.
Understanding the differences between healthy and AA HSPCs could uncover the cause of the T cell targeting.
However HSPCs are severely depleated in AA patients, making them a difficult target to study \citep{zhu_single-cell_2021}.
Recent studies like \citet{tonglin_single-cell_2022} and \citet{zhu_single-cell_2021} have used single-cell RNA-sequencing (scRNA-seq) to perform differential expression analysis between healthy and AA patient HSPCs.
These studies have thus far used PCA to reduce dataset dimensionality, followed by graph-based clustering and likelihood ratio tests to identify significantly differentially expressed genes between clusters \citep{zhu_single-cell_2021}.
These studies have successfully identified differentially expressed genes between healthy and AA HSPCs.
However, their adjustments for noise and batch effects are limited to their initial dimension reductions.
This initial dimension reduction assumes a generalized linear model is sufficient for accurately mapping onto the low-dimensional manifold underlying the data.
This is a particularly problematic assumption in papers like \citet{tonglin_single-cell_2022} where samples were only taken from four patients (two healthy and two with AA).
With such a low n of patients, it would be challenging to convincingly claim differences between AA and healthy clusters are due to disease status as opposed to being due to differences between individuals.

In this project we explored methods of learning differences between healthy and AA HSPCs using the scRNA-seq data published by \citet{tonglin_single-cell_2022}.
We used computational methods to model both healthy and diseased HSPC transcriptome profiles with the goal of identifying perturbations capable of moving a diseased HSPC to a healthy cell or vice versa.
Such identified perturbations could both suggest potential new therapies and help identify the mechanisms of AA onset and treatment.
We have not yet reached this goal of identifying perturbations.
However, our work sheds light on what methods are useful when working with data collected from a very small number of individuals.


\section{Methods and abbreviated results}

\subsection{The Data}
For this study we used the scRNA-seq data published by \citet{tonglin_single-cell_2022}.
These data were from bone marrow samples taken from two healthy donors and two AA patients.
These data comprise 20419 HSPCs (NUMBER AA and NUMBER healthy) with 33538 genes per cell.
Sequencing was performed by 10x Genomics \citep{10X_genomics}.

\subsection{Data Preprocessing}
We downloaded the data from the NCBI Gene Expression Omnibus (https://www.ncbi.nlm.nih.gov/geo/ , GEO accession:GSE181989).
We used the python package Scanpy \citep{wolf_scanpy_2018} to filter out all genes that [INSERT GENE THRESHOLD] and all cells that [INSERT CELL THRESHOLD].
After filtering, we had [NUMBER] cells with [NUMBER] genes.
[NUMBER] of these cells were from the two AA patients and [NUMBER] of these cells were from the healthy controls.
DID WE DO ANY IMPUTATION HERE?

\subsection{Principal Component and UMAP Visualizations}
% In accordance with standards in the field, we wanted to visualize the data using principal component (PC) projection and UMAP.
We used the Scanpy python package \citep{wolf_scanpy_2018} to make these visualizations.
For the PC projection, we projected the data onto the first two principal components which captured 10.6\% and 2.7\% of the overall variance of the dataset respectively.
We colored each cell by disease status (see Figure \ref{PCA_projections_disease}) and by individual (see Figure \ref{PCA_projections_individual}).

\subsection{Linear Discriminant Analysis}
We used the Scanpy python package \citep{wolf_scanpy_2018} to perform linear discriminant analysis (LDA) on the first 50 principal components of our data.
We used 5-fold cross validation to assess the model's prediction accuracy.
Our model accurately predicted the disease status of 98.2\% of the cells in the training data and 98.0\% of the cells in the test data.

To assess whether LDA was distinguishing between transcriptome profiles of diseased vs healthy individuals or just differentiating between the individuals in general, we retrained the model using data from just two individuals (one sick and one diseased).
We then used this model to predict the disease status of the other two individuals.
We started by again using the first 50 principal components.
Our model accurately predicted the disease status 48.5\% of the test cells.
We switched the individuals that were used to train/test the model and found the resulting model accurately predicted the disease status of 67\% of the test cells.
We also performed this analysis using the first 5 principal components, which yielded 56.6\% and 60.4\% testing accuracy for each training/test data pair.

\subsection{scVI}
We used the scvi-tools python package to perform scVI \citep{lopez_deep_2018, gayoso_python_2022}.
INSERT INFORMATION ABOUT THE TECNICAL DETAILS HERE.


% TODO: Put these in the correct locations and make the captions fit the figures.
\begin{figure}
  \centering
  \includegraphics[width=0.5\textwidth]{disease_status.png}
  \caption{Principal Component and UMAP Projection visualizations colored by disease status. 0 is healthy, 1 is AA.}
  \label{PCA_projections_disease}
\end{figure}

\begin{figure}
  \centering
  \includegraphics[width=0.5\textwidth]{individual.png}
  \caption{Principal Component and UMAP Projection visualizations colored by individual. Individuals 0 and 1 have AA. Individuals 2 and 3 are healthy controls.}
  \label{PCA_projections_individual}
\end{figure}


\section{Discussion and Results}

\subsection{Visualization reveals cells do not cluster by disease status}
Existing AA scRNA-seq data analysis uses linear dimensionality reduction techniques as a preprocessing step prior to clustering.
We began our analysis by following these same steps and assessing whether such preprocessing was sufficient for generating a dataset capable of training a simple but generalizable linear classifier.
We first visualized our data following a tutorial provided by the Scanpy[CITE] python package and qualitatively noticed that the diseased and healthy cells do not seem to cluster distinctly (see Figure \ref{PCA_projections_disease}).
We also noticed that cells from different individuals seemed to have very different transcriptional profiles (see Figure \ref{PCA_projections_individual}).
This led us to hypothesize that a linear model would either perform poorly when predicting disease status or perform well by learning to distinguish between individuals instead of distinguishing between disease status.

\subsection{LDA is insufficient for predicting disease status}
We performed five-fold cross validation on an LDA model trained on the first 50 principal components of the dataset and found the prediction accuracy to be an outlandishly high 98\%.
We subsequently hypothesised that this accuracy could be attributed to the model learning to distingish each individual as opposed to the model distinguishing between healthy and AA individuals.
To test this hypothesis, we used the same model hyperparameters but trained our model on data from exclusively one healthy and one AA individual.
Our hypothesis was that if the model is truly learning to distinguish between illness status, a model trained on data from representative AA patients and healthy individuals should be able to accurately predict the disease status of cells from other AA and healthy patients.
Conversely, if the model is learning to distingish individuals, a model trained on some individuals should not accurately predict the disease status of other individuals.
Because our dataset only contains data from four individuals, we trained the model on data from two individuals (one sick and one AA) and tested the model using the other two patients' data.
We then swapped the test and the training data and again assessed prediction accuracy.
Our measured prediction accuracy was 48.5\% and 67\%.
This decrease in accuracy supported our hypothesis that, even though it may be able to distinguish individuals, a simple linear model is not sufficient to reliably distinguish between AA and healthy cells.

We next considered whether larger principal components contained the dimensions along which the AA and healthy cells separated most distinctly and whether small variance introduced by the smaller principal components may be making the task of performing LDA more challenging.
To investigate this possibility, we ran this analysis again using only the first five principal components of the dataset.
This analysis yielded a prediction accuracies of 56.6\% and 60.4\%, leading us to conclude that even the largest principal components encompass variances unrelated to disease status.

\subsection{scVI does something I hope}
We discovered scVI through our class reading and wanted to investigate its performance on the \citet{tonglin_single-cell_2022} data.
We trained scVI on the \citet{tonglin_single-cell_2022} data using the scvi-tools python packge \citep{gayoso_python_2022} with the intention of exploring the differences in AA cell distributions vs healthy cell distributions in latent space.
We next wanted to build a simple classifier using this latent representation in order to create a path between the two cell-type distributions.
% - OR use pretrained scvi, transfer to AA data -- find bigger data of blood cells train SCVI, transfer to AA data real quick
% - OR use someone else's actual pretrained scvi
% 1. simple interpolation between samples from centroids of data domains
% 2. build simple classifier from latent representations -> use to create a path from sample to altered sample (like in original plan) 


\section{Conclusion}
scRNA-seq data is challenging to analyze due to the inherent noise, batch effects, dropout rates, and differences between individuals.
The challenge of accounting for person-to-person variability is amplified when there are a small number of individuals in your dataset.
The \citet{tonglin_single-cell_2022} data gave us an opportunity to experiment with a low-individual scRNA-seq dataset while working towards answering mechanistic questions about a devistating disease.
Our investigations revealed that simple linear models are insufficient for discriminating between disease states with such a low number of individuals.
Something about scVI.


{
\small
\printbibliography
}


\end{document}
